Updating the README to reflect 2020 situation (#22)
nodes-ios/Playbook: README.md

- encouraging our team members to join iOS related meetups, groups and conferences
- encouraging our team members to speak at iOS events
- contributing back to some of the other open source libraries we use through issues and PRs
- encouraging our team members to write tech articles about any topic they desire on our [engineering blog](https://engineering.nodesagency.com/); if needed, we can also suggest topics for new articles

> We create lots of apps for our clients. Most of those apps rely heavily on web APIs. We abstracted the common operations for working with web APIs, and that's how [Serpent](https://github.com/nodes-ios/Serpent) (formerly known as Serializable) and [Cashier](https://github.com/nodes-ios/Cashier) were born. Another issue that was common in all the apps we do was handling keyboard appearance events. This led to the appearance of [KeyboardHelper](https://github.com/nodes-ios/KeyboardHelper).

Reviewer: Maybe this should not mention all the repo's and isntead point to an org with all the OSS frameworks? :)

Author: It's not all the repos, there are way more on `nodes-ios` :). I think it gives a nice story of how some of our most used libraries came to be. If you want, I can remove it and just say something like 

We also make open source stuff sometimes. You can find it [here](https://github.com/nodes-ios/).

But I think that sounds a bit dull.

Reviewer: Fair, leave it there :) It’s also not my decision to make on this, better to ask @chriscombs  😇

# Team meetings
Here are the recurring meetings the iOS team takes part in:

#### iOS Team Meeting
This is the main meeting in the iOS team, where we talk about the latest things happening in our team. General direction of the team, brief general team performance review, what people need help with, amount of workload, should we use or not _that_ framework, etc. This happens in the first Monday of each month. 

#### Friday Tech Talk
This is a monthly meeting, that occurs on the fourth Friday of every month. Anyone in the engineering team can talk about anything cool that they implemented recently. It could be using some new technology, some cool framework, a nice animation, a clever algorithm or anything similar to this. The meeting is held in the Copenhagen office, but if anyone from other offices wants to attend, we can enable a video call.

#### Monthly Briefing

> This is a monthly meeting, where the management tells all the employees how the company did in the last month and what our targets and short, mid and long-term plans are. Topics covered are sales, projects, cases, tech, events, staff and others. The meeting is held in the Copenhagen office, but other offices can join in by video call.

Reviewer: Not sure this should be included in iOS playbook.

Author: Personally, I'd like to read that in the iOS Playbook in my first day at Nodes, or on the public playbook of any company I'd consider joining. But maybe that's just me. Indeed, it's not platform specific. Here I just updated the old info for the quarterly catch-up from years ago. 
Let me know if you want me to delete it or not.

Reviewer: Sure, but it's not applied the same globally. Let's leave it for now and we can handle this in the future.

- Get another iOS developer to invite you to our developer accounts.
- We want you to help us make great open source software. So get another iOS developer to invite you to  [our GitHub organisation](https://github.com/nodes-ios). Make sure you enable your 2FA on GitHub. Our client projects are on [our private git](https://github.com/nodes-projects) and our public ones are on [GitHub](https://github.com/nodes-ios).
- Set up your git keys properly, so you can pull and push from both git servers.
- You should have been invited to our Slack, JIRA, and Postman accounts. Ask IT if you don't have access.
- Make sure to read the [Nodes Handbook](https://docs.google.com/document/d/1E4ZyGqIKDttGlJ0c0rEkNy2lxP-n1PBwQzuaESbLnpw/edit) (private document). That is meant for all employees, and tells a bit of our history, how we got here, but also how we handle different tasks.
- We use [Happeo](https://app.happeo.com/home) for internal (semi)official announcements. Ask IT if you don't have access.

# Tools We Use

> For employee management we use [BambooHR](https://nodesagency.bamboohr.com/home). Ask IT if ou don't have access and the Chief People Officer if anything in there is unclear. Use this to request time off, register sick day and manage your personal info and banking info for salary pay.

Reviewer 2: ```suggestion
For employee management we use [BambooHR](https://nodesagency.bamboohr.com/home). Ask IT if you don't have access and the Chief People Officer if anything in there is unclear. Use this to request time off, register sick day and manage your personal info and banking info for salary pay.
```

Our main communication tool is [**Slack**](https://nodes.slack.com). We use this for daily communication.

Our main project management tool is [**JIRA**](https://nodesagency.atlassian.net/jira/). Each project has its own JIRA board. Always keep your project's JIRA board up to date to reflect the current status of the project. We work in an agile manner and move fast, but the JIRA board must clearly reflect the project's status. Ask [Simon](https://nodes.slack.com/messages/@siej) or your project manager if you have questions.

> With the same JIRA account you will also have access to Confluence, where you can find project-related documentation, and also internal docuements.

Reviewer: ```suggestion
With the same JIRA account, you will also have access to Confluence, where you can find project-related documentation, and also internal documents.
```

> [**Harvest**](https://nodes.harvestapp.com/) is the time tracking tool that we use. It helps us see how much time one person has worked on a project. Make sure to always harvest on the appropriate project. Ask the PM on which Harvest project you should track the time. We expect you to harvest 7.5 hours per day. Lunch time is not harvested, and it doesn't count towards the 7.5 hours per day. And if you want more detailed instructions on how to harvest, ask [Simon](https://nodes.slack.com/messages/@siej). You can also set up auto-harvesting in you Ournodes acount and it will automatically harvest 7.5 hours per day on the project youre assigned to.

Reviewer: This is not true in all offices, Berlin harvest 8 hours.

Author: Should I remove the fluffy text and just link to the Harvest section in the handbook? This way, we won't need to care about updating the playbook when/if the harvesting rules change.

Reviewer: Well, I don't really mind however it's not the same across all offices :) And if this is supposed to be the global playbook it will not apply to many places.

Author: I updated it with an `(8 in Germany)`, removed all other mentions (I hope) of 7.5 hours and linked to the handbook instead of redirecting people to Simon :)

[**Postman**](https://www.getpostman.com/) is a great tool which helps you see and test web APIs. Our backend team uses Postman to test and document their APIs, so all the web APIs for our apps can be found in Postman. You can use it to see the different endpoints available, read their documentation or make requests to those endpoints. Ask IT if you don't have access.

Our design team uses Sketch. But you, as a developer, don't need to have Sketch installed. To ease the collaboration between designers and developers, we use [**Zeplin**](https://zeplin.io/). In Zeplin, you can see each screen in the design, get info about the sizes, padding, fonts, colours and also export assets for the mobile devices. You can also add notes in Zeplin and communicate with your designer directly on the project.

[**Charles**](https://www.charlesproxy.com/) is an HTTP proxy. You can set your phone / simulator to proxy through Charles and you can see all the API calls it made, you can inspect its requests or responses. It comes in very handy especially if you work with an external, poorly documented API. Ask IT if you need a license.

> For in-house distribution and crash reporting, we use [**Firebase**](firebase.google.com/). Depensing on the project, sometimes we only use TestFlight. Talk to your PM or with previous developers on that project.

Reviewer: ```suggestion
For in-house distribution and crash reporting, we use [**Firebase**](firebase.google.com/). Depending on the project, sometimes we only use TestFlight. Talk to your PM or with previous developers on that project.
```

> We had been using [**Carthage**](https://github.com/Carthage/Carthage) for a long time to manage our dependencies on iOS. We chose Carthage over CocoaPods because for us, this is the one that brought the most advantages. But since we started using Firebase more and more and Google pushes strongly for CocoaPods, we sometimes use both in the same project. We appreciate all the effort put into Carthage, as well as into CocoaPods, and we're looking for the right time to switch to Swift Package Manager as our default dependency manager for iOS projects.

Reviewer: ```suggestion
We had been using [**Carthage**](https://github.com/Carthage/Carthage) for a long time to manage our dependencies on iOS. We chose Carthage over CocoaPods because, for us, this is the one that brought the most advantages. But since we started using Firebase more and more and Google pushes strongly for CocoaPods, we sometimes use both in the same project. We appreciate all the effort put into Carthage, as well as into CocoaPods, and we're looking for the right time to switch to Swift Package Manager as our default dependency manager for iOS projects.
```

> The way we do localisation in Nodes is a bit different than in other places. We use [**NStack**](https://nstack.io/), a service we built, which together with the [**NStack SDK**](https://github.com/nstack-io/nstack-ios-sdk) offers dynamic localisation for our apps. Go to [nstack.io](https://nstack.io/), log in with your Nodes account, select the app you need (or create a new one), go to "Translate" and add new translations or edit the current ones. A translation consists of a key (by which the string is recognised through the SDK) and one or more values (depending on the number of languages your app is translated to). The translations can be changed in NStack and the changes will reflect in the app without the need of an app update.

Reviewer: ```suggestion
The way we do localization in Nodes is a bit different than in other places. We use [**NStack**](https://nstack.io/), a service we built, which together with the [**NStack SDK**](https://github.com/nstack-io/nstack-ios-sdk) offers dynamic localization for our apps. Go to [nstack.io](https://nstack.io/), log in with your Nodes account, select the app you need (or create a new one), go to "Localize" and add new translations or edit the current ones. A translation consists of a key (by which the string is recognized through the SDK) and one or more values (depending on the number of languages your app is translated to). The translations can be changed in NStack and the changes will reflect in the app without the need for an app update.
```

# Your First Day

> 

Reviewer: I don't think we need to remove this, I just think we shouldn't specify the people :)

Reviewer 3: > I've removed a line related to onboarding which I think was very specific to iOS devs in Copenhagen; I'm not familiar with `halp`, I don't think I was introduced to it yet 🙈, so maybe I'm not the best person to write about it.

Halp is just a slack command `/halp` for getting help from IT

Author: Yeah, I had removed it because I wasn't sure how onboarding for iOS devs in other offices goes. Do they have an online meeting with the head of ios?

Reviewer: Yes, that's how it works everywhere as far as I know.
